renku/api/models/run.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017-2020 - Swiss Data Science Center (SDSC)
+# A partnership between École Polytechnique Fédérale de Lausanne (EPFL) and
+# Eidgenössische Technische Hochschule Zürich (ETHZ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""API Workflow Models."""
+
+from pathlib import Path
+
+from renku.api.models.project import ensure_project_context
+from renku.core.models.cwl.command_line_tool import get_indirect_inputs_path, get_indirect_outputs_path
+
+
+class _PathBase:
```

Comment:
Another thought that comes to mind is that we can extend this to support non-local/non-file inputs at somepoint, like `renku.api.S3Bucket` that streams data. As opposed to/in addition to the pointer file approach we discussed at some point.

Reply:
Good point! I prefer to implement `os.PathLike` because if users try to get auto-completion on an Input/Output object, they won't see lots of other methods from `pathlib.Path`.

Reply:
> Another thought that comes to mind is that we can extend this to support non-local/non-file inputs at somepoint, like renku.api.S3Bucket that streams data. As opposed to/in addition to the pointer file approach we discussed at some point.

This would be a great use-case if there is a library that provides streaming access to S3.